src/test/java/tdl/s3/sync/destination/DestinationWithCounter.java

```diff
@@ -0,0 +1,106 @@
+package tdl.s3.sync.destination;
+
+import ch.qos.logback.classic.Logger;
+import ch.qos.logback.classic.LoggerContext;
+import ch.qos.logback.classic.spi.LoggingEvent;
+import com.amazonaws.services.s3.model.PartETag;
+import com.amazonaws.services.s3.model.PartListing;
+import com.amazonaws.services.s3.model.UploadPartRequest;
+import java.util.List;
+import org.slf4j.LoggerFactory;
+import tdl.s3.log.RequestLogAppender;
+import tdl.s3.upload.MultipartUploadResult;
+
+public class DestinationWithCounter implements Destination {
+
+    private final S3BucketDestination destination;
+
+    private int count = 0;
+
+    public DestinationWithCounter(S3BucketDestination destination) {
+        this.destination = destination;
+    }
+
+    public int getCount() {
+        return count;
+    }
+
+    public RequestLogAppender getLog() {
+        Logger logbackLogger = (Logger) LoggerFactory.getLogger(org.slf4j.Logger.ROOT_LOGGER_NAME);
+        LoggerContext lc = logbackLogger.getLoggerContext();
+        Logger rootLogger = lc.getLogger(Logger.ROOT_LOGGER_NAME);
+        return (RequestLogAppender) rootLogger.getAppender("REQUEST");
+    }
+
+    private String createLogCheckpoint() {
+        RequestLogAppender log = getLog();
+        if (log == null) {
+            return "";
+        }
+        return log.createCheckpoint();
+    }
+
+    private int getHttpSizeByCheckpoint(String checkpoint) {
+        RequestLogAppender log = getLog();
+        if (log == null) {
+            return 0;
+        }
+        List<Object> list = log.getLogListFromCheckpoint(checkpoint);
+        if (list == null) {
+            return 0;
+        }
+        int size = 0;
+        //TODO; to stream
+        for (Object object : list) {
+            if (!(object instanceof LoggingEvent)) {
+                continue;
+            }
+            LoggingEvent e = (LoggingEvent) object;
+            if (!e.getLoggerName().startsWith("org.apache.http")) {
+                continue;
+            }
+            System.out.println(e.getMessage());
+            size += e.getMessage().length();
```

Comment:
@julianghionoiu the idea is that we can parse the wire log message and determine the size.